src/queue.c

```diff
@@ -6216,10 +6243,61 @@ _dispatch_worker_thread(void *context)
 	_dispatch_sigmask();
 #endif
 	_dispatch_introspection_thread_add();
+	dispatch_priority_t pri = dq->dq_priority;
+	pthread_priority_t pp = _dispatch_get_priority();
+
+#if defined(__linux__)
```

Comment:
Can this be `defined(__unix__)` instead of `defined(__linux__)`? Is there anything in this block that is Linux-specific?

Reply:
Yes. the behaviour of `setpriority` is specific to linux and not applicable to other UNIX-like systems.